content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1099 @@
+im = Image.open(filename)
+```
+
+Here is the test image we will send to the Deployed Model
+
+```python
+im
+```
+
+![png](/img/output_26_0.png)
+
+Here we will encode the image into into base64 binary format 
+
+```python
+image = open(filename, 'rb')  # open binary file in read mode
+image_read = image.read()
+image_64_encode = base64.b64encode(image_read)
+bytes_array = image_64_encode.decode('utf-8')
+request = {
+  "instances": [
+    {
+      "data": bytes_array
+    }
+  ]
+}
+result_file = "{filename}.{ext}".format(filename=str(filename).split(".")[0], ext="json")
+print("Result File: ",result_file)
+with open(result_file, 'w') as outfile:
+    json.dump(request, outfile, indent=4, sort_keys=True)
+```
+
+```
+Result File:  kserve_utils/torchserve_utils/example_img.json
+```
+
+Lets submit the image to our deployed endpoint and visualize the predictions!
+
+```python
+headers = {
+    "Host": "torchserve.default.example.com"
+}
+
+data = open(result_file)
+response = requests.post('http://localhost:8080/v1/models/xview-fasterrcnn:predict', headers=headers, data=data)
+
+resp = json.loads(response.content)
+
+print("Number of Predictions: ", len(resp['predictions'][0]))
+```
+
+```
+Number of Predictions:  95
+```
+
+```python
+draw = ImageDraw.Draw(im)
+```
+
+```python
+for pred in resp['predictions'][0]:
+    assert len(list(pred.keys())) == 2
+    cl_name = list(pred.keys())[0]
+    bboxes = pred[cl_name]
+    # print(cl_names)
+    # print(bboxes)
+    # print("score: ",pred['score'])
+    if pred['score'] > 0.4:
+        # bboxes = [int(i) for i in bboxes]
+        # print(bboxes[0],type(bboxes[0]))
+        draw.rectangle([bboxes[0],bboxes[1],bboxes[2],bboxes[3]],outline=(255,0,0),fill=None,width=1)
+        draw.text([bboxes[0],bboxes[1]-10],"{} :{:.2f}".format(cl_name,pred['score']),fill=(250,0,0))
+```
+
+Here we will see the model predictions overlaid on the input image
+
+```python
+plt.figure(figsize=(12,12))
+plt.imshow(im)
+```
+
+```
+<matplotlib.image.AxesImage at 0x7ff6341ccf10>
+```
+
+![png](/img/output_34_1.png)
+
+## Conclusion
+
+There are numerous ways to get started with end-to-end ML model training with HPE. You can find the [full example on GitHub here](https://github.com/interactivetech/e2e_blogposts/tree/main/ngc_blog); to get started with Determined AI’s open source model training platform, visit the [Documentation](https://docs.determined.ai/latest/) page.  
+
+If you’re ready to begin your ML journey with HPE, we’re excited to help you get started! [HPE Machine Learning Development Environment](https://www.hpe.com/us/en/solutions/artificial-intelligence/machine-learning-development-environment.html) comes with premium HPE support, which you can read more about [here](https://www.hpe.com/us/en/solutions/artificial-intelligence/machine-learning-development-environment.html).  
+
+HPE also offers a purpose-built, turnkey AI infrastructure for model development and training \
```

Comment:
Please remove the "backslash" character at the end of the sentence: HPE also offers a purpose-built, turnkey AI infrastructure for model development and training.